R/source.R

```diff
@@ -429,33 +434,30 @@ check_cargo_output <- function(compilation_result, message_buffer, tty_has_color
     jsonlite::parse_json
   )
 
-  if (!isTRUE(quiet)) {
-    purrr::walk(
-      gather_cargo_output(
-        cargo_output,
-        "warning",
-        tty_has_colors
-      ),
-      ui_w
-    )
-  }
-
   if (!isTRUE(compilation_result$status == 0)) {
-    error_messages <- purrr::map_chr(
-      gather_cargo_output(
-        cargo_output,
-        "error",
-        tty_has_colors
-      ),
-      bullet_x
+    error_messages <- gather_cargo_output(
+      cargo_output,
+      "error",
+      tty_has_colors
     )
 
-    ui_throw(
-      "Rust code could not be compiled successfully. Aborting.",
+    errors_formatted <- purrr::map_chr(
       error_messages,
+      cli::format_inline,
+      keep_whitespace = TRUE
+    ) %>%
+      # removing double new lines with single new line
+      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
+      # ensures that the leading cli style `x` is there
+      rlang::set_names("x")
+
+    rlang::abort(
+      c(
+        "Rust code could not be compiled successfully. Aborting.",
+        unlist(errors_formatted)
+      ),
       call = call,
-      glue_open = "{<{",
-      glue_close = "}>}"
+      class = "rextendr_error"
```

Comment:
Could this whole expression be one pipeline? The temporary vars don't seem necessary.

Reply:
I suppose? I'm not sure how I would do it. It might look something like the below. I'm not sure if it is any better?

```r
  purrr::map_chr(
  error_messages,
  cli::format_inline,
  keep_whitespace = TRUE
) %>%
  # removing double new lines with single new line
  stringi::stri_replace_all_fixed("\n\n", "\n") %>%
  # ensures that the leading cli style `x` is there
  rlang::set_names("x") %>%
  { c("Rust code could not be compiled successfully. Aborting.", .)} %>%
  rlang::abort(class = "rextendr_error")
```

Reply:
Though I suppose the error message part can be put into a single one along the lines of below which feels reasonable

```r
error_messages <- gather_cargo_output(
      cargo_output,
      "error",
      tty_has_colors
    ) %>% purrr::map_chr(
      cli::format_inline,
      keep_whitespace = TRUE
    ) %>%
      # removing double new lines with single new line
      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
      # ensures that the leading cli style `x` is there
      rlang::set_names("x")

    rlang::abort(
      c(
        "Rust code could not be compiled successfully. Aborting.",
        errors_messages
      ),
      call = call,
      class = "rextendr_error"
    )
```

Reply:
I prefer this. @Ilia-Kosenkov can you make a call please?

Reply:
+1 to pipeline approach

Reply:
(although I note you're missing the final pipe in the above example)

Reply:
In which way? It cannot be directly piped to `rlang::abort()` without first adding 
```r
... above pipeline ... %>%
  { c( "Rust code could not be compiled successfully. Aborting.", .) } %>%
  rlang::abort(call = call, class = "rextendr_error")
```

I'm of the mind that if you have to use inline braces in a pipe then it probably ought to be broken. Though this is y'all work not mine! I defer.

Reply:
Erm, I suppose I can make it work with `%>% append("Rust code could not be compiled successfully. Aborting", .)`

Reply:
stretching my brain!! resolving with `append(..., after = 0)`

Reply:
What's wrong with:
```R
gather_cargo_output(
      cargo_output,
      "error",
      tty_has_colors
    ) %>% purrr::map_chr(
      cli::format_inline,
      keep_whitespace = TRUE
    ) %>%
      # removing double new lines with single new line
      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
      c(
            "Rust code could not be compiled successfully. Aborting.",
            x = errors_messages
       ) %>%
        rlang::abort(
            call = call,
           class = "rextendr_error"
       )
```

Reply:
Honestly, I would say that this is an overkill. It is ok to build error messages using pipe, but I see not particular usefulness in going this extra mile. Negatively affects readability.

Reply:
Oh, I find this readable but happy for it to stay as two expressions.

Reply:
This is where it is at right now 

https://github.com/JosiahParry/rextendr/blob/d3227398750e6eb708744333c3a815c3dd4aab94/R/source.R#L453

Reply:
```r
    gather_cargo_output(
      cargo_output,
      "error",
      tty_has_colors
    ) %>%
      purrr::map_chr(
        cli::format_inline,
        keep_whitespace = TRUE
      ) %>%
      # removing double new lines with single new line
      stringi::stri_replace_all_fixed("\n\n", "\n") %>%
      # ensures that the leading cli style `x` is there
      rlang::set_names("x") %>%
      # prepend the actual error message
      append("Rust code could not be compiled successfully. Aborting.", after = 0) %>%
      rlang::abort(
        call = call,
        class = "rextendr_error"
    )
 ```

Reply:
I find `c`  bit clearer because you can set names. But anyway Ilia doesn't like that so maybe just revert to the two expression option.

Reply:
Resolved.